src/main/scala/monix/kafka/KafkaConsumer.scala

```diff
@@ -0,0 +1,140 @@
+package monix.kafka
+
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Ack.Stop
+import monix.execution.FutureUtils.extensions.FutureExtensions
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.{Observable, Observer}
+import monix.reactive.observers.Subscriber
+import org.apache.kafka.clients.consumer.{ConsumerRecord, ConsumerRecords, KafkaConsumer => ApacheKafkaConsumer}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+import scala.collection.JavaConverters._
+
+/** An `Observable` implementation that reads messages from `Kafka`.
+  *
+  * You need to call `addTopicSubscription` in order for the Observable to start sending messages from that topic.
+  */
+final class KafkaConsumer[K, V] private
+  (config: KafkaConsumerConfig, ioScheduler: Scheduler)
+  (implicit K: Deserializer[K], V: Deserializer[V])
+  extends Observable[ConsumerRecord[K, V]] with StrictLogging {
+
+  def unsafeSubscribeFn(subscriber: Subscriber[ConsumerRecord[K, V]]): Cancelable = {
+    val source = underlying.onErrorHandleWith { ex =>
+      logger.error(
+        s"Unexpected error in KafkaConsumerObservable, retrying in ${config.retryBackoffTime}", ex)
+      // Applying configured delay before retry!
+      underlying.delaySubscription(config.retryBackoffTime)
+    }
```

Comment:
This error recovery made sense in our code, but in this public library if it ends in error, then that's the right behavior. Users are then free to apply `onErrorHandleWith` on their own. We shouldn't be so eager about logging either.

Or in other words the `underlying` observable should probably be this subscribe function.